pkg/deploy/lattice/target_group_manager.go

```diff
@@ -320,69 +277,36 @@ func (s *defaultTargetGroupManager) List(ctx context.Context) ([]targetGroupOutp
 	return tgList, err
 }
 
-func isNameOfTargetGroup(targetGroup *model.TargetGroup, name string) bool {
-	if targetGroup.Spec.Config.IsServiceImport {
-		// We are missing protocol info for ServiceImport, but we do know the RouteType.
-		// Relying on the assumption that we have one TG per (RouteType, Service),
-		// do a simple guess to find the matching TG.
-		validProtocols := []string{
-			vpclattice.TargetGroupProtocolHttp,
-			vpclattice.TargetGroupProtocolHttps,
-		}
-		validProtocolVersions := []string{
-			vpclattice.TargetGroupProtocolVersionHttp1,
-			vpclattice.TargetGroupProtocolVersionHttp2,
-		}
-		if targetGroup.Spec.Config.ProtocolVersion == vpclattice.TargetGroupProtocolVersionGrpc {
-			validProtocolVersions = []string{vpclattice.TargetGroupProtocolVersionGrpc}
-		}
-
-		for _, p := range validProtocols {
-			for _, pv := range validProtocolVersions {
-				candidate := &model.TargetGroup{
-					Spec: model.TargetGroupSpec{
-						Name: targetGroup.Spec.Name,
-						Config: model.TargetGroupConfig{
-							Protocol:        p,
-							ProtocolVersion: pv,
-						},
-					},
-				}
-				if name == getLatticeTGName(candidate) {
-					return true
-				}
-			}
-		}
-		return false
-	} else {
-		return name == getLatticeTGName(targetGroup)
-	}
-}
-
 func (s *defaultTargetGroupManager) findTargetGroup(
 	ctx context.Context,
-	targetGroup *model.TargetGroup,
+	modelTargetGroup *model.TargetGroup,
 ) (*vpclattice.TargetGroupSummary, error) {
-	vpcLatticeSess := s.cloud.Lattice()
-	targetGroupListInput := vpclattice.ListTargetGroupsInput{}
-	resp, err := vpcLatticeSess.ListTargetGroupsAsList(ctx, &targetGroupListInput)
+	listInput := vpclattice.ListTargetGroupsInput{}
+	resp, err := s.cloud.Lattice().ListTargetGroupsAsList(ctx, &listInput)
 	if err != nil {
```

Comment:
Nit:
Seems we have 2 places to invoke findTargetGroup():
- (s *defaultTargetGroupManager) Upsert()
- (s *defaultTargetGroupManager) Delete()

I assmue above 2 places just manage "local" cluster's target groups(i.e., intendToCreate/intendToDelete TG.VpcId === config.VpcID)?

In that case, can we add a `VpcIdentifier` filtering condition when listTG to improve preformance?:
```
listInput := vpclattice.ListTargetGroupsInput{
VpcIdentifier: &config.VpcID,
}
resp, err := s.cloud.Lattice().ListTargetGroupsAsList(ctx, &listInput)
```
(Could consider to do It in a seperate PR)

Reply:
I'll double-check, but this should work. I didn't realize we had Vpc as an input to the list call, thanks for calling this out!

Reply:
Having a closer look it's possible that in the service export case the VPC will be different than what's in the controller's config. Using the vpc id of the target group we're looking for should work though.